pkg/image/resolve_test.go

```diff
@@ -191,6 +192,69 @@ func TestResolveWithCluster(t *testing.T) {
 	}
 }
 
+func TestGetImages(t *testing.T) {
+	tests := []struct {
+		name         string
+		exportConfig ExportConfig
+		expected     []string
+		notExpected  []string
+	}{
+		{
+			name: "exportConfig is completely empty",
+			exportConfig: ExportConfig{
+				SystemChartsPath: "",
+				ChartsPath:       "",
+				OsType:           Linux,
+				RancherVersion:   "",
+				GithubEndpoints:  []GithubEndpoint{},
+			},
+			expected:    []string{},
+			notExpected: []string{"rancher/ui-plugin-catalog"},
+		},
+		{
+			name: "only extensions is set in exportConfig",
+			exportConfig: ExportConfig{
+				SystemChartsPath: "",
+				ChartsPath:       "",
+				OsType:           Linux,
+				RancherVersion:   "",
+				GithubEndpoints:  ExtensionEndpoints,
```

Comment:
It may not be a good idea to add this unit test as per my understanding when it runs it actually makes HTTP calls to `https://api.github.com/repos/rancher/ui-plugin-charts/releases` which by definition makes it not a unit test as there is a dependency on external system outside of immediate `rancher/rancher` codebase. We may also get rate-limited by GH if this test runs a lot.

Is this something we already do in other unit tests in `rancher/rancher`?

Mocking HTTP server may be a way to go but not sure how involved it will be.

(The first test case is fine though)

Reply:
Done. I have resolved it by using setupTestServer which was already used in other unit tests.

Reply:
Perfect, thank you!